### PR TITLE
fixes #3172 Reconcile > Use values as identifiers" does not reconcile #4655 #4664

### LINKED ISSUE #3172: "Reconcile > Use values as identifiers" does not reconcile
### To Reproduce
Run "Reconcile" > "Use values as identifiers" on any column, whether it contains properly-formatted unique IDs for the given service or not. You will see that it seems not to reconcile at all - it only says, whatever your cell contents are, it's 100% matched and generates a real-looking URL, whether or not that URL will resolve. 

### Expected Behavior
Reconciliation seems to be missing the part where it actually validates these unique IDs to see if they match up with existing entities. I expect failures to say so. 

### Screenshots
error1

I tested on Wikidata and VIAF. VIAF gives no hover-information, the matches just send you to 404s, e.g. http://viaf.org/viaf/Q17291. None of the content in that VIAF column should be matching (Q### is an invalid format) except that first value, which resolves correctly to https://viaf.org/viaf/38242123/.  

There are two things happening on Wikidata that I thought I'd mention - IDs that don't exist give the error in the above screenshot. IDs that seem to exist but have yet to be assigned (something that fits the Q### format) look a bit different (so that may be a second bug to work out). I would expect these to have some other kind of obvious error message or flag. 
error2

### Versions
Windows, Firefox
 - OpenRefine: the new 3.4 release, as well as 3.4 beta 2 (last week)

### PR DESCRIPTION
**Pull request by WaltonG**
Fixes #3172

Changes proposed in this pull request:
- Notice added when selecting service

### CODE REVIEW

**Comment by WaltonG** [comment]
@wetneb Pull request https://github.com/OpenRefine/OpenRefine/pull/4666 is open for review

**Comment by WaltonG** [comment]
Working on it right now

File: main/webapp/modules/core/scripts/views/data-table/menu-reconcile.js [hunk 1]
```diff
@@ -141,6 +141,7 @@ DataTableColumnHeaderUI.extendMenu(function(column, columnHeaderUI, menu) {
     frame.width("400px");
 
     var header = $('<div></div>').addClass("dialog-header").text($.i18n('core-views/use-values-as-identifiers/header')).appendTo(frame);
+    var header2 = $('<div></div>').addClass("dialog-header").text($.i18n('core-views/use-values-as-identifiers-note')).appendTo(frame);
```
[the remark this answers is not in the document]
Comment (WaltonG) [reply main/webapp/modules/core/scripts/views/data-table/menu-reconcile.js:144]: Am working on it

[the remark this answers is not in the document]
Comment (WaltonG) [reply main/webapp/modules/core/scripts/views/data-table/menu-reconcile.js]: I have actioned as reviewed

**Comment by WaltonG** [comment]
@elroykanye Check out this PR. Deploy has been cancelled

**Comment by elroykanye** [comment]
> @elroykanye Check out this PR. Deploy has been cancelled

It should be running up now.

**Comment by wetneb** [comment]
Deploying is only done when you make changes to the docs.

File: main/webapp/modules/core/langs/translation-en.json [hunk 2]
```diff
@@ -738,6 +738,7 @@
     "core-views/replacement-info": "If \"regular expression\" option is checked and finding pattern contains groups delimited with parentheses, $0 will return the complete string matching the pattern, and $1, $2… the 1st, 2nd… group.",
     "core-views/replace-dont-escape": "use \\n for new lines, \\t for tabulation, \\\\n for \\n, \\\\t for \\t.",
     "core-views/warning-regex": "Invalid regular expression : $1",
+    "core-views/use-values-as-identifiers-note": "Reconciliation identifiers will not be validated",
```
Comment (wetneb) [inline main/webapp/modules/core/langs/translation-en.json:741]: Perhaps a final period could be nice?

**Comment by WaltonG** [comment]
> Deploying is only done when you make changes to the docs.

I will proceed to make changes in the docs

File: main/webapp/modules/core/langs/translation-en.json (hunk 2 above)
Comment (WaltonG) [reply main/webapp/modules/core/langs/translation-en.json:741]: True. Am on it.

**Comment by antoine2711** [comment]
@WaltonG : I think this PR should also take care of the preview when there is no data return, and show a cleaner explanation then a JSON string… What do you think?

(From the issue)
image

Regards, Antoine

**Comment by WaltonG** [comment]
@antoine2711 A text explanation will provide a  better user experience. Am working on it

**Comment by WaltonG** [comment]
@antoine2711 I have tried reconciliation with different services, Wikidata, GND, VIAF, OpenData, and ORCID. Each service has a different response as reflected in the preview
1. Wikidata - {"arguments":{"id":"93","lang":"en"},"details":"'93'","message":"invalid query","status":"error"}
2. GND - Not found : 93
3. VIAF- No preview
4. OpenData - No preview
5. ORCID - No preview
In the case of Wikidata and GND or any other services that choose to return a json response I will find it challenging to determine if the json response contains meaningful data or an error message. Could there be a way around this ?

**Comment by ostephens** [comment]
@WaltonG @antoine2711 I think dealing with the response here is a separate issue. Looking at #3172 the priority is identified as telling the user that when they choose to use identifiers as reconciled, there is no actual validation. In the issue @wetneb  writes

> We could:
> 
> - Add text to the dialog where we choose the reconciliation service to use, explaining that identifiers will not be validated;
> - Improve the user experience after that, improving the way we handle invalid ids (for instance in the Wikidata extension)

But emphasises that only the first of these is the 'good first issue' aspect. I think we should focus on that in this PR, and separate out dealing with incorrect/invalid responses separately as it's quite a bit more complex.

**Review by wetneb — APPROVED** [review]

**Pull request merged**